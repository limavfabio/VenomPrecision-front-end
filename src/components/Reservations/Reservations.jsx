import React, { useEffect, useState } from "react";
import { FaPlaneDeparture } from "react-icons/fa";
import { GiModernCity } from "react-icons/gi";
import { MdProductionQuantityLimits } from "react-icons/md";
import { useDispatch, useSelector } from "react-redux";
import { fetchReservations } from "../../redux/reservationSlice";
import Header from "../layout/Header";
import MobileSidebar from "../layout/MobileSidebar";
import Sidebar from "../layout/Sidebar";

const Reservations = () => {
  const dispatch = useDispatch();
  const userId = useSelector((state) => state.user.id);
  const { reservations } = useSelector((state) => state.reservations.value);
  useEffect(() => {
    dispatch(fetchReservations(parseInt(userId, 10)));
  }, [dispatch, userId]);

  // State for sidebar
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
  console.log(userId, reservations);
  const toggleSidebar = () => {
    setIsSidebarOpen(!isSidebarOpen);
  };
  return (
    <>
      <Header toggleSidebar={toggleSidebar} />
      <div className="flex">
        <Sidebar />
        {isSidebarOpen && <MobileSidebar />}

        <div className="h-screen w-full flex-1 overflow-y-auto bg-gray-200 pt-[3rem]">
          <h2 className="my-5 text-center text-3xl font-bold uppercase">
            My Reservations
          </h2>
          <section className="bg-blueGray-50 ">
            <div className="mx-auto mb-12 mt-5 w-full px-10 xl:mb-0">
              <div className="relative mb-6 flex w-full min-w-0 flex-col break-words rounded  shadow-lg ">
                <div className="block w-full overflow-x-auto">
                  <table className="w-full border-collapse items-center  rounded border bg-transparent">
                    <thead className="bg-gray-500 text-white">
                      <tr>
                        <th className="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase">
                          <span className="inline-block pr-2 align-middle text-xl">
                            <GiModernCity />
                          </span>
                          City
                        </th>
                        <th className="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase">
                          <span className="inline-block pr-2 align-middle text-xl">
                            <MdProductionQuantityLimits />{" "}
                          </span>
                          Product
                        </th>
                        <th className="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase">
                          <span className="inline-block pr-2 align-middle text-xl">
                            <FaPlaneDeparture />{" "}
                          </span>{" "}
                          Visits
                        </th>
                      </tr>
                    </thead>
                    <tbody className="bg-white">
                      {reservations &&
                      Array.isArray(reservations) &&
                      reservations.length > 0 ? (
                        reservations.map((item) => (
                          <tr key={item.reserver_at}>
                            <th className="text-blueGray-700 whitespace-nowrap border-l-0 border-r-0 border-t-0 p-4 px-6 text-left align-middle text-xs">
                              {item.city}
                            </th>
                            <td className="whitespace-nowrap border-l-0 border-r-0 border-t-0 p-4 px-6 align-middle text-xs">
                              {item.product_name}
                            </td>
                            <td className="whitespace-nowrap border-l-0 border-r-0 border-t-0 p-4 px-6 align-middle text-xs">
                              {item.date}
                            </td>
                          </tr>
                        ))
                      ) : (
                        <tr>
                          <td colSpan="4" className="px-6 py-4 text-center">
                            Reservations not found
                          </td>
                        </tr>
                      )}
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </>
  );
};

export default Reservations;
