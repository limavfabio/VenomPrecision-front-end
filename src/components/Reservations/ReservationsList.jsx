import React from 'react';

const ReservationsList = ({ reservations }) => (
  <div>
    {reservations
      && Array.isArray(reservations)
      && reservations.length > 0 ? (
        reservations.map((item) => (
          <tr key={item.reserver_at}>
            <th className="text-blueGray-700 whitespace-nowrap border-l-0 border-r-0 border-t-0 p-4 px-6 text-left align-middle text-xs">
              {item.city}
            </th>
            <td className="whitespace-nowrap border-l-0 border-r-0 border-t-0 p-4 px-6 align-middle text-xs">
              {item.product_name}
            </td>
            <td className="whitespace-nowrap border-l-0 border-r-0 border-t-0 p-4 px-6 align-middle text-xs">
              {item.date}
            </td>
          </tr>
        ))
      ) : (
        <tr>
          <td colSpan="4" className="px-6 py-4 text-center">
            Reservations not found
          </td>
        </tr>
      )}
  </div>
);

export default ReservationsList;
