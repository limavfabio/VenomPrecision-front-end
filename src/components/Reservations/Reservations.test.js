import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Reservations from './Reservations';

// Create a mock store function
const mockStore = configureStore();

// Mock the react-router-dom module
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockReturnValue({ pathname: '/mock-path' }),
  useNavigate: jest.fn(),
}));

describe('Reservations test', () => {
  test('renders the component without errors', () => {
    const initialState = {
      user: {
        id: 1,
      },
      reservations: {
        value: [],
      },
    };
    const store = mockStore(initialState);

    render(
      <Provider store={store}>
        <Reservations />
      </Provider>
    );

    // Add your assertions here to verify the component behavior

    const tableElement = screen.getByRole("table");
    expect(tableElement).toBeInTheDocument();
  });
});
