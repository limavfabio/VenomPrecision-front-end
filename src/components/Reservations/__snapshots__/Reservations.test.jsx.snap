// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Home expects to render a snapshot of the Home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen w-full flex-1 overflow-y-auto bg-gray-200 pt-[3rem]"
      >
        <h2
          class="my-5 text-center text-3xl font-bold"
        >
          MY RESERVATIONS
        </h2>
        <section
          class="bg-blueGray-50 "
        >
          <div
            class="mx-auto mb-12 mt-5 w-full px-10 xl:mb-0"
          >
            <div
              class="relative mb-6 flex w-full min-w-0 flex-col break-words rounded  shadow-lg "
            >
              <div
                class="block w-full overflow-x-auto"
              >
                <table
                  class="w-full border-collapse items-center  rounded border bg-transparent"
                >
                  <thead
                    class="bg-gray-500 text-white"
                  >
                    <tr>
                      <th
                        class="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase"
                      >
                        <span
                          class="inline-block pr-2 align-middle text-xl"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 512 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M217 25v30h30V25h-30zm48 0v30h30V25h-30zm48 0v30h30V25h-30zm-96 48v46h30V73h-30zm48 0v46h30V73h-30zm48 0v46h30V73h-30zm-96 64v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zm174 6.508l-94 53.715V215h94v-71.492zM217 201v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zm80 32v110h94V233h-94zM39 256v23h18v-23H39zm178 9v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM37.562 297l-7 14h146.875l-7-14H37.563zM25 329v158h23v-23h32v23h48v-23h32v23h23V329H25zm192 0v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM48 352h32v16H48v-16zm80 0h32v16h-32v-16zm265 9v126h31v-23h32v23h31V361h-94zM48 384h32v16H48v-16zm80 0h32v16h-32v-16zm89 9v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM48 416h32v16H48v-16zm80 0h32v16h-32v-16zm89 41v30h30v-30h-30zm48 0v30h30v-30h-30zm48 0v30h30v-30h-30z"
                            />
                          </svg>
                        </span>
                        City
                      </th>
                      <th
                        class="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase"
                      >
                        <span
                          class="inline-block pr-2 align-middle text-xl"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M13 10h-2V8h2v2zm0-4h-2V1h2v5zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03L21 4.96 19.25 4l-3.7 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
                            />
                          </svg>
                           
                        </span>
                        Product
                      </th>
                      <th
                        class="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase"
                      >
                        <span
                          class="inline-block pr-2 align-middle text-xl"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 640 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M624 448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h608c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM80.55 341.27c6.28 6.84 15.1 10.72 24.33 10.71l130.54-.18a65.62 65.62 0 0 0 29.64-7.12l290.96-147.65c26.74-13.57 50.71-32.94 67.02-58.31 18.31-28.48 20.3-49.09 13.07-63.65-7.21-14.57-24.74-25.27-58.25-27.45-29.85-1.94-59.54 5.92-86.28 19.48l-98.51 49.99-218.7-82.06a17.799 17.799 0 0 0-18-1.11L90.62 67.29c-10.67 5.41-13.25 19.65-5.17 28.53l156.22 98.1-103.21 52.38-72.35-36.47a17.804 17.804 0 0 0-16.07.02L9.91 230.22c-10.44 5.3-13.19 19.12-5.57 28.08l76.21 82.97z"
                            />
                          </svg>
                           
                        </span>
                         
                        Visits
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="bg-white"
                  >
                    <tr>
                      <td
                        class="px-6 py-4 text-center"
                        colspan="4"
                      >
                        Reservations not found
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen w-full flex-1 overflow-y-auto bg-gray-200 pt-[3rem]"
    >
      <h2
        class="my-5 text-center text-3xl font-bold"
      >
        MY RESERVATIONS
      </h2>
      <section
        class="bg-blueGray-50 "
      >
        <div
          class="mx-auto mb-12 mt-5 w-full px-10 xl:mb-0"
        >
          <div
            class="relative mb-6 flex w-full min-w-0 flex-col break-words rounded  shadow-lg "
          >
            <div
              class="block w-full overflow-x-auto"
            >
              <table
                class="w-full border-collapse items-center  rounded border bg-transparent"
              >
                <thead
                  class="bg-gray-500 text-white"
                >
                  <tr>
                    <th
                      class="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase"
                    >
                      <span
                        class="inline-block pr-2 align-middle text-xl"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 512 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M217 25v30h30V25h-30zm48 0v30h30V25h-30zm48 0v30h30V25h-30zm-96 48v46h30V73h-30zm48 0v46h30V73h-30zm48 0v46h30V73h-30zm-96 64v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zm174 6.508l-94 53.715V215h94v-71.492zM217 201v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zm80 32v110h94V233h-94zM39 256v23h18v-23H39zm178 9v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM37.562 297l-7 14h146.875l-7-14H37.563zM25 329v158h23v-23h32v23h48v-23h32v23h23V329H25zm192 0v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM48 352h32v16H48v-16zm80 0h32v16h-32v-16zm265 9v126h31v-23h32v23h31V361h-94zM48 384h32v16H48v-16zm80 0h32v16h-32v-16zm89 9v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM48 416h32v16H48v-16zm80 0h32v16h-32v-16zm89 41v30h30v-30h-30zm48 0v30h30v-30h-30zm48 0v30h30v-30h-30z"
                          />
                        </svg>
                      </span>
                      City
                    </th>
                    <th
                      class="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase"
                    >
                      <span
                        class="inline-block pr-2 align-middle text-xl"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                          <path
                            d="M13 10h-2V8h2v2zm0-4h-2V1h2v5zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03L21 4.96 19.25 4l-3.7 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
                          />
                        </svg>
                         
                      </span>
                      Product
                    </th>
                    <th
                      class="bg-blueGray-50 text-blueGray-500 border-blueGray-100 whitespace-nowrap border border-l-0 border-r-0 border-solid px-6 py-3 text-left align-middle text-xs font-semibold uppercase"
                    >
                      <span
                        class="inline-block pr-2 align-middle text-xl"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 640 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M624 448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h608c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM80.55 341.27c6.28 6.84 15.1 10.72 24.33 10.71l130.54-.18a65.62 65.62 0 0 0 29.64-7.12l290.96-147.65c26.74-13.57 50.71-32.94 67.02-58.31 18.31-28.48 20.3-49.09 13.07-63.65-7.21-14.57-24.74-25.27-58.25-27.45-29.85-1.94-59.54 5.92-86.28 19.48l-98.51 49.99-218.7-82.06a17.799 17.799 0 0 0-18-1.11L90.62 67.29c-10.67 5.41-13.25 19.65-5.17 28.53l156.22 98.1-103.21 52.38-72.35-36.47a17.804 17.804 0 0 0-16.07.02L9.91 230.22c-10.44 5.3-13.19 19.12-5.57 28.08l76.21 82.97z"
                          />
                        </svg>
                         
                      </span>
                       
                      Visits
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="bg-white"
                >
                  <tr>
                    <td
                      class="px-6 py-4 text-center"
                      colspan="4"
                    >
                      Reservations not found
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
